enhancements/storage/storage-hypershift.md

```diff
@@ -0,0 +1,1050 @@
+cluster.
+
+The components currently are:
+
+* cluster-csi-snapshot-controller-operator and its operand, csi-snapshot-controller.
+* cluster-storage operator and its operand aws-ebs-csi-driver-operator and its operand AWS EBS CSI driver.
+  * All other CSI driver operators + CSI drivers should follow the same approach, AWS EBS is just the first one to
+    run its control plane in HyperShift's management cluster.
+  
+## Motivation
+
+A guest cluster in HyperShift should ideally run only user workloads, and it should not be possible to break the cluster
+from within it. [See HyperShift goals](https://hypershift-docs.netlify.app/reference/goals-and-design-invariants/).
+In this enhancement, we're getting closer to this goal, however, an evil cluster-admin can still break storage parts of
+the guest cluster. See Risks and Mitigations below.
+
+### User Stories
+
+See [HyperShift Personas](https://hypershift-docs.netlify.app/reference/concepts-and-personas/) for definition of
+various roles used below.
+
+* As a Cluster Instance Admin / User, I want the cluster to run as little OCP infrastructure workloads as possible to
+  have the cluster just for my applications.
+
+* As a Cluster Instance Admin, I want the cluster to be as robust as possible, I should not be able to break it
+  by my actions.
+  
+* As a Cluster Instance Admin / User, I don't want OCP infrastructure components running there to have elevated RBACs,
+  so a compromised component cannot break my cluster (too much).
+  * Still, CSI driver Pods *must* run on each worker node in the guest cluster as privileged, to be able to format and
+    mount volumes. A compromised CSI driver Pod can do anything to the application data - steal it, delete it, encrypt
+    it..., and it can fully compromise the node.
+    
+* As a Cluster Service Consumer, I want to spawn new OCP guest clusters on AWS, so the guest clusters can use AWS EBS
+  storage out of the box.
+  * Explicitly, the guest clusters can run in different AWS regions and zones than the HyperShift management cluster.
+    The management cluster does not need to run on AWS at all. In addition, one management cluster may run guest
+    clusters in different AWS regions or even in different clouds.
+
+* As a Cluster Service Consumer, I want my OCP guest clusters with no worker nodes, so I can 1) autoscale the cluster
+  from zero or 2) "hibernate" the cluster when it's idle and resume it later.
+  * Control-plane components should keep working without any issues.
+    * Cluster Instance Users should be e.g. able to delete PVCs and expect that the corresponding volume is deleted in AWS.
+    * Dynamic provisioning should work if forced by `volumeBindingMode: Immediate`. Users are then responsible
+      for creating nodes in the right availability zones so that they can use the provisioned volumes.
+      
+* Phase 2: As a Cluster Service Consumer, I want to get metrics of storage management cluster components, so I can
```

Comment:
As a Cluster Service Provider...

The service provider is responsible for the SLO on storage in this model, not the consumer.